app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -174,10 +174,8 @@ protected function prepareOptionIds(array $optionsArray)
     {
         if (isset($optionsArray['value']) && is_array($optionsArray['value'])) {
             foreach (array_keys($optionsArray['value']) as $optionId) {
-                if (isset($optionsArray['delete']) && isset($optionsArray['delete'][$optionId])) {
-                    if ($optionsArray['delete'][$optionId]== 1) {
-                        unset($optionsArray['value'][$optionId]);
-                    }
+                if (isset($optionsArray['delete'][$optionId]) & $optionsArray['delete'][$optionId]) {
```

Comment:
Sure ! 
It's not a double condition (&&) but a binary operator, like the "!" you use with empty. 
Nano second, your right, but it's more than nothing, I think the code is readable :)